tasks/docker.py

```diff
@@ -11,6 +11,7 @@
 
 # Order matters here
 RELEASE_CONTAINERS = [
+    "cpp-root",
```

Comment:
Does this need to be here? This is the containers that get rebuilt for every release. `cpp-root` just contains dependencies and no application code. Additionally, the Docker release build is heavyweight enough as it is, I'd rather not introduce more steps.

Reply:
I agree that it's a bit pointless to push it, especially if you have it on your system already. My intention was more to make sure the `cpp-root` container would get built next time you run the release command. This `release` command is reserved for you anyway so I agree I shouldn't touch it.

I will instead make a separate "build all containers" command. Which will allow me for example to test changes to them or allow new-commers to reliably test them.

As for the speed issue since it seems that it is still not possible to have parallel builds in Docker I propose to you to move it to Github Actions that will be triggered on tagged builds (i.e. only on master)? Either running the normal runner, or making a self-hosted runner on one of the Koalas machine?

Reply:
`cpp-root` is purely used to capture bits of the `base` container that don't often get rebuilt, but are expensive to build when they do. As a result, I'm happy to leave its rebuilding to be ad-hoc for now and just focus on the stuff that really needs to get rebuilt (i.e. the release containers). I'm not sure a "build all containers" command is very useful, and would inevitably not get maintained (i.e. new containers would accidentally get left off the list). Even when developing `cpp-root` surely you'd just be building the `cpp-root` container or `base` (in which case you can run `inv docker.build -c cpp-root -c base`. If you think it would be useful then go for it, but I don't think I would ever use it.